src/character.cpp

```diff
@@ -2052,3 +2055,145 @@ long Character::ammo_count_for( const item &gun )
 
     return ret;
 }
+
+void Character::deal_melee_hit( Creature *source, int hit_spread, bool crit,
+                                const damage_instance &d, dealt_damage_instance &dealt_dam )
+{
+    Creature::deal_melee_hit( source, hit_spread, crit, d, dealt_dam );
+    const int total_damage = dealt_dam.total_damage();
+    if( total_damage > 0 ) {
+        practice( skill_dodge, 5 * total_damage, 10 );
+    }
+}
+
+void Character::deal_projectile_attack( Creature *source, dealt_projectile_attack &attack )
+{
+    Creature::deal_projectile_attack( source, attack );
+    const int total_damage = attack.dealt_dam.total_damage();
+    if( total_damage > 0 ) {
+        practice( skill_dodge, 5 * total_damage, 10 );
```

Comment:
Amount should suffice as a temporary measure until we get a better implementation of dodges/blocks